presto-main/src/main/java/com/facebook/presto/sql/planner/PartitioningScheme.java

```diff
@@ -22,17 +22,54 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.IntPredicate;
 
+import static com.facebook.presto.sql.planner.PartitioningScheme.Replication.REPLICATE_NOTHING;
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class PartitioningScheme
 {
+    public enum Replication
+    {
+        REPLICATE_NOTHING(
```

Comment:
The more I think about it, the more convinced I am that this commit is unnecessary (at least, for the purpose of fixing the "nulls on left side of semijoin issue"). At the end of this PR, we still have two possible values for the enum (NOTHING vs NULLS_AND_ANY). We could just rename the `replicateNulls` variable and be done with it.